okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -72,20 +81,32 @@ static RealCall newRealCall(OkHttpClient client, Request originalRequest, boolea
       executed = true;
     }
     captureCallStackTrace();
+    timeout.enter();
     eventListener.callStart(this);
     try {
       client.dispatcher().executed(this);
       Response result = getResponseWithInterceptorChain();
       if (result == null) throw new IOException("Canceled");
       return result;
```

Comment:
Should you call timeout.exit() here on the happy path?

Reply:
We don’t exit until the response body is exhausted, which could occur after this returns.